pkgs/tools/audio/beets/default.nix

```diff
@@ -39,6 +39,7 @@ lib.makeExtensible (self: {
       hash = "sha256-yWwxYSzSSmx2UfCn0EBH23hQGZKSRn/c8ryvxLUeHdM=";
     };
     pluginOverrides = {
+      # unstable has a new plugin, so we register it here.
```

Comment:
I don't understand, if you register it, why don't you also `enable = true;`, like in the examples from the comment above?

Reply:
Because it defaults to `enable = !disableAllPlugins`.

That's what you want for builtin plug-ins. When you explicitly want to make sure something will be enabled, you set it to `enable = true` explicitly.

Reply:
I don't think this distinction really matters, we can iterate on it in the future.

Do you have any blocking concerns?

Reply:
A plugin has attrs. The attrs are `enable`, `propagatedBuildInputs`, and `wrapperBins`. If you do not specify an attr for a plugin, a default value will be given. The default value for `enable` is `!disableAllPlugins`.

If you do `pluginOverrides = { foo = { }; }` that will end up in the list of all plugins as:
```
{
  # ...
  foo = {
    enable = !disableAllPlugins;
    propagatedBuildInputs = [];
    wrapperBins = [];
  };
  # ...
}
```

The default value of `!disableAllPlugins` is `false`, so nothing being changed, `foo` will be enabled.

If you, for example, wanted to enable _only_ the fetchart plugin, you could do this:
```Nix
beets.override {
  disableAllPlugins = true;
  pluginOverrides.fetchart.enable = true;
};
```

Because you will override the `enable` value of `fetchart` from `!disableAllPlugins` to `true`.

IOW: Setting to `true` just makes it disobey `disableAllPlugins` and force-enables it.

Reply:
> IOW: Setting to `true` just makes it disobey `disableAllPlugins` and force-enables it.

I understand now :) Thanks for the explanation.